src/filter.rs

```diff
@@ -75,6 +76,10 @@ impl Filter {
             }
           })
       }
+      Self::Save { path } => match state.save(path.clone()) {
```

Comment:
Can you take a reference here in stead of cloning?

```suggestion
      Self::Save { path } => match state.save(&path) {
```

Reply:
`path` is already a reference and `state::save` takes in an owned value, so we do actually need to clone it here